synapse/storage/databases/main/schema/delta/58/19txn_id.sql

```diff
@@ -0,0 +1,32 @@
+/* Copyright 2020 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+-- A map of recent events persisted with transaction IDs. Used to deduplicate
+-- send event requests with the same transaction ID.
+--
+-- Note, transaction IDs are scoped to the user ID/access token that was used to
+-- make the request.
+CREATE TABLE event_txn_id (
+    event_id TEXT NOT NULL,
+    user_id TEXT NOT NULL,
+    token_id BIGINT NOT NULL,
```

Comment:
I've made these `ON DELETE CASCADE`, rather than adding explicit handling when we delete. I think this is fine because this is mostly a temporary cache of values, and if a someone deletes an event/access token we shouldn't be trying to de-duplciate?

Reply:
seems fine.